### ISSUE TITLE
NoClassDefFoundError: java/sql/Timestamp

### ISSUE THREAD

**Issue by daschneider** [oracle/visualvm#123]
When trying to do JDBC profiling the process being profiled generates the following error (many times):

lang.NoClassDefFoundError - java/sql/Timestamp Details: java/sql/Timestamp
java.lang.NoClassDefFoundError: java/sql/Timestamp
	at org.netbeans.lib.profiler.server.ProfilerRuntimeCPU.converToString(ProfilerRuntimeCPU.java:788) ~[?:?]
	at org.netbeans.lib.profiler.server.ProfilerRuntimeCPU.addParameter(ProfilerRuntimeCPU.java:687) ~[?:?]
	at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.setTimestamp(HikariProxyPreparedStatement.java) ~[HikariCP-3.2.0.jar:?]

Perform a JDBC profile of an application that executes a lot of database statements.

**Expected behavior**
Should be able to generate a JDBC profile.

**Desktop (please complete the following information):**
 - OS: Ubuntu server 18.04
 - JDK version [11.0.1+13]
 - Version [1.4.2]
 - Database PostgreSQL 10.x

**Comment by thurka** [comment]
It looks like this is caused by JDK modularisation in JDK 9+. Can you try to run your application on JDK 8 instead of JDK 11?

**Comment by daschneider** [comment]
Unfortunately we can't run our app using Java 8.  The app uses the new HTTP 2 client classes from Java 11 so it has to run with Java 11 or newer.

**Comment by thurka** [comment]
Hopefully fixed in master.

**Comment by daschneider** [comment]
Thanks for the quick turn around :)

**Comment by MOHIT-PRASANTH** [comment]
I am using 1.4.4 version, facing the same issue while running JDBC profiling.
Caused by: java.lang.NoClassDefFoundError: java/sql/Timestamp

**Comment by thurka** [comment]
> I am using 1.4.4 version, facing the same issue while running JDBC profiling.
> Caused by: java.lang.NoClassDefFoundError: java/sql/Timestamp

This is strange. Please file separate issue. Do not forget to attach messages.log and full exception. Thanks.

**Comment by airdrik** [comment]
I've run into this as well, and seeing that the code in question is in the Netbeans codebase, I went ahead and reported the issue upstream: https://issues.apache.org/jira/browse/NETBEANS-3318

**Comment by thurka** [comment]
No, it does not. From Java point of view, there is no difference between GraalVM and JVM.

Can you, please, post the stacktrace?

**Comment by airdrik** [comment]
Basically the same as the OP:
```
java.lang.NoClassDefFoundError: java/sql/Date
         org.netbeans.lib.profiler.server.ProfilerRuntimeCPU.converToString(ProfilerRuntimeCPU.java:768)
         org.netbeans.lib.profiler.server.ProfilerRuntimeCPU.addParameter(ProfilerRuntimeCPU.java:663)
         org.apache.commons.dbcp2.DelegatingPreparedStatement.setDate(DelegatingPreparedStatement.java:378)
         io.tekniq.jdbc.TqConnectionExtKt.applyParams(TqConnectionExt.kt:80)
... <snip additional lines of application code>

```

note: I get this from the application being profiled when VisualVM (or NetBeans' profiler) is hooked up to profile the application (a Tomcat app running on Java 11).

**Comment by airdrik** [comment]
The change made here applied to the copy of ProfilerRuntimeCPU in the `org.graalvm.visualvm.lib.jfluid.server` package, whereas the error is comming from the copy in the ` org.netbeans.lib.profiler.server` namespace.

I have verified that downloading the NetBeans code, applying the patch you have here to that code and rebuilding, then copying the built libprofilerinterface.so files into the appropriate locations in VisualVM's profiler/lib/deployed fixes the issue.

**Comment by thurka** [comment]
This stack trace is from Apache NetBeans and does not contain the fix. Can you please, _restart_ Tomcat process and use VisualVM only. It looks like NetBeans' profiling agent was deployed first and used by subsequently by VisualVM.

**Comment by airdrik** [comment]
Okay, I shutdown NetBeans and restarted tomcat; (reinstalled VisualVM to revert the "fix" I'd applied), and now it works.  
Thank you for pointing out about the interactions between the different profiling agents, that was helpful.

**Comment by thurka** [comment]
> Okay, I shutdown NetBeans and restarted tomcat; (reinstalled VisualVM to revert the "fix" I'd applied), 

Note that `libprofilerinterface.so` does not contain `ProfilerRuntimeCPU` bytecode. It is native part of profiler agent and does not contain any classes. So `ProfilerRuntimeCPU` was used from VisualVM 1.4.4.

>and now it works.

Thanks for confirmation that it is fixed in VisualVM 1.4.4. 

> Thank you for pointing out about the interactions between the different profiling agents, that was helpful.

Funny part is that after you deployed fixed agent from VisualVM, NetBeans will also work OK.
